ui/client/common/ProcessStateUtils.js

```diff
@@ -1,39 +1,124 @@
+    FINISHED: "The process completed successfully.",
+  },
+
+  STATUS_ICONS: {
+    UNKNOWN: InlinedSvgs.iconUnknown,
+    NOT_DEPLOYED: InlinedSvgs.iconNotDeployed,
+    DURING_DEPLOY: InlinedSvgs.iconDeployRunningAnimated,
+    DEPLOYED: InlinedSvgs.iconDeploySuccess,
+    RUNNING: InlinedSvgs.iconRunningAnimated,
+    CANCELED: InlinedSvgs.iconStoppingSuccess,
+    RESTARTING: InlinedSvgs.iconStoppedWorking,
+    FAILED: InlinedSvgs.iconFailed,
+    ERROR: InlinedSvgs.iconStoppedWorking,
+    FINISHED: InlinedSvgs.iconSuccess,
+  },
+
+  isRunning(state) {
+    return this.getStateStatus(state) === this.STATUSES.RUNNING
+  },
 
   isDeployed(process) {
-    const action = _.get(process, 'deployment.action')
-    return action != null && action.toUpperCase() === this.ACTION_DEPLOY
+    return this.getProcessAction(process) === this.ACTIONS.DEPLOY
+  },
+
+  getStateTooltip(state) {
+    return this.getStatusTooltip(this.getStateStatus(state))
+  },
+
+  getStatusTooltip(status) {
+    return _.get(this.STATUS_TOOLTIPS, status, this.STATUS_TOOLTIPS[this.STATUSES.UNKNOWN])
+  },
+
+  getStateIcon(state) {
+    return this.getStatusIcon(this.getStateStatus(state))
   },
 
-  isCanceled(deployment) {
-    const action = _.get(deployment, 'deployment.action')
-    return action != null && action.toUpperCase() === this.ACTION_CANCEL
+  getStatusIcon(status) {
+    return _.get(this.STATUS_ICONS, status, this.STATUS_ICONS[this.STATUSES.UNKNOWN])
   },
 
-  //TODO: display e.g. warnings, use it in Visualization (left panel)
-  getStatusClass(processState, shouldRun, statusesLoaded) {
-    const isRunning = _.get(processState, 'isRunning')
-    if (isRunning) {
-      return "status-running"
-    } else if (shouldRun) {
-      return statusesLoaded ? "status-notrunning" : "status-unknown"
+  getStateStatus(state) {
+    const status = _.get(state, 'status', null)
+
+    if (_.isUndefined(state) || (state && status === null)) {
+      return this.STATUSES.UNKNOWN
+    }
+
+    //TODO: In future it will be nice when API return state with status CANCELED
+    if (_.isNull(state)) {
+      return this.STATUSES.CANCELED
     }
-    return null
+
+    return state.status.toUpperCase()
   },
 
-  getStatusMessage(processState, shouldRun, loaded) {
-    const isRunning = _.get(processState, 'isRunning')
-    if (isRunning) {
-      return "Running"
-    } else if (shouldRun) {
-      const message = (processState || {}).errorMessage || "Not found in engine"
-      return loaded ? `Not running: ${message}` : "Unknown state"
+  getProcessStateStatus(process, state) {
+    if (state == null) {
+      return this.getProcessStatus(process)
     }
 
-    return null
+    return this.getStateStatus(state)
+  },
+
+  getProcessAction(process) {
+    return _.get(process, 'deployment.action', null)
+  },
+
+  getProcessTooltip(process) {
+    return this.getStatusTooltip(this.getProcessStatus(process))
+  },
+
+  getProcessStatus(process) {
```

Comment:
I map deployment information from process to status, because I in next step I map status to icon. I want normalize getting icon / tooltip to getting it by status.